sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OpenAiController.java

```diff
@@ -54,7 +58,26 @@ ResponseEntity<String> chatCompletion(
   @Nonnull
   ResponseEntity<ResponseBodyEmitter> streamChatCompletionDeltas() {
     final var message = "Can you give me the first 100 numbers of the Fibonacci sequence?";
-    return service.streamChatCompletionDeltas(message);
+    final var stream = service.streamChatCompletionDeltas(message);
+    final var emitter = new ResponseBodyEmitter();
+    final Runnable consumeStream =
+        () -> {
+          final var totalOutput = new OpenAiChatCompletionOutput();
+          // try-with-resources ensures the stream is closed
+          try (stream) {
+            stream
+                .peek(totalOutput::addDelta)
+                .forEach(delta -> send(emitter, delta.getDeltaContent()));
+          } finally {
+            send(emitter, "\n\n-----Total Output-----\n\n" + objectToJson(totalOutput));
+            emitter.complete();
```

Comment:
(Comment)
It would be cool if Spring Boot had an easier way to deal with non-blocking responses 🙄